This is ready for review, but I don't know why OfBorg isn't happy with this; in particular, it's finding a link failure, which I can't reproduce locally building `/nix/store/bl3xsj6mym955b1v9d45yy2w4xgccgvs-envoy-1.19.1.drv`:

```
ERROR: /build/source/source/exe/BUILD:23:16: Linking of rule '//source/exe:envoy-static' failed (Exit 1): gcc failed: error executing command /nix/store/xiq6j4jsyj351p8q3yw9cg1hdqp9m685-gcc-wrapper-10.3.0/bin/gcc @bazel-out/k8-opt/bin/source/exe/envoy-static-2.params
[4,995 / 4,995] checking cached actions
external/com_google_cel_cpp/parser/parser.cc:155: error: undefined reference to 'cel_grammar::CelLexer::~CelLexer()'
external/com_google_cel_cpp/parser/parser.cc:157: error: undefined reference to 'cel_grammar::CelParser::~CelParser()'
external/com_google_cel_cpp/parser/parser.cc:155: error: undefined reference to 'cel_grammar::CelLexer::CelLexer(antlr4::CharStream*)'
external/com_google_cel_cpp/parser/parser.cc:157: error: undefined reference to 'cel_grammar::CelParser::CelParser(antlr4::TokenStream*)'
external/com_google_cel_cpp/parser/parser.cc:176: error: undefined reference to 'cel_grammar::CelParser::start()'
external/com_google_cel_cpp/parser/parser.cc:157: error: undefined reference to 'cel_grammar::CelParser::~CelParser()'
external/com_google_cel_cpp/parser/parser.cc:155: error: undefined reference to 'cel_grammar::CelLexer::~CelLexer()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/parser/parser.o(.debug_addr+0xc08): error: undefined reference to 'cel_grammar::CelParser::start()'
external/com_google_cel_cpp/parser/visitor.cc:134: error: undefined reference to 'cel_grammar::CelParser::StartContext::expr()'
external/com_google_cel_cpp/parser/visitor.cc:102: error: undefined reference to 'cel_grammar::CelParser::PrimaryExprContext::primary()'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::NestedContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::IdentOrGlobalCallContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::CreateListContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::CreateStructContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::ConstantLiteralContext'
external/com_google_cel_cpp/parser/visitor.cc:120: error: undefined reference to 'cel_grammar::CelParser::MemberExprContext::member()'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::PrimaryExprContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::SelectOrCallContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::IndexContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::CreateMessageContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::StartContext'
external/com_google_cel_cpp/parser/visitor.cc:134: error: undefined reference to 'cel_grammar::CelParser::StartContext::expr()'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::ExprContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::ConditionalAndContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::ConditionalOrContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::RelationContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::CalcContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::LogicalNotContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::PrimaryExprContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::MemberExprContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::SelectOrCallContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::MapInitializerListContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::NegateContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::IndexContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::UnaryContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::CreateListContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::CreateMessageContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::CreateStructContext'
external/com_google_cel_cpp/parser/visitor.cc:361: error: undefined reference to 'cel_grammar::CelParser::ConstantLiteralContext::literal()'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::IntContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::UintContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::DoubleContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::StringContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::BytesContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::BoolFalseContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::BoolTrueContext'
external/com_google_cel_cpp/parser/visitor.cc:46: error: undefined reference to 'typeinfo for cel_grammar::CelParser::NullContext'
external/com_google_cel_cpp/parser/visitor.cc:263: error: undefined reference to 'cel_grammar::CelParser::IndexContext::member()'
external/com_google_cel_cpp/parser/visitor.cc:249: error: undefined reference to 'cel_grammar::CelParser::SelectOrCallContext::member()'
external/com_google_cel_cpp/parser/visitor.cc:231: error: undefined reference to 'cel_grammar::CelParser::LogicalNotContext::member()'
external/com_google_cel_cpp/parser/visitor.cc:234: error: undefined reference to 'cel_grammar::CelParser::LogicalNotContext::member()'
external/com_google_cel_cpp/parser/visitor.cc:240: error: undefined reference to 'cel_grammar::CelParser::NegateContext::member()'
external/com_google_cel_cpp/parser/visitor.cc:243: error: undefined reference to 'cel_grammar::CelParser::NegateContext::member()'
external/com_google_cel_cpp/parser/visitor.cc:189: error: undefined reference to 'cel_grammar::CelParser::RelationContext::calc()'
external/com_google_cel_cpp/parser/visitor.cc:190: error: undefined reference to 'cel_grammar::CelParser::RelationContext::calc()'
external/com_google_cel_cpp/parser/visitor.cc:198: error: undefined reference to 'cel_grammar::CelParser::RelationContext::relation(unsigned long)'
external/com_google_cel_cpp/parser/visitor.cc:200: error: undefined reference to 'cel_grammar::CelParser::RelationContext::relation(unsigned long)'
external/com_google_cel_cpp/parser/visitor.cc:207: error: undefined reference to 'cel_grammar::CelParser::CalcContext::unary()'
external/com_google_cel_cpp/parser/visitor.cc:208: error: undefined reference to 'cel_grammar::CelParser::CalcContext::unary()'
external/com_google_cel_cpp/parser/visitor.cc:216: error: undefined reference to 'cel_grammar::CelParser::CalcContext::calc(unsigned long)'
external/com_google_cel_cpp/parser/visitor.cc:218: error: undefined reference to 'cel_grammar::CelParser::CalcContext::calc(unsigned long)'
external/com_google_cel_cpp/parser/visitor.cc:271: error: undefined reference to 'cel_grammar::CelParser::CreateMessageContext::member()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x580): error: undefined reference to 'cel_grammar::CelParser::CreateMessageContext::member()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x1770): error: undefined reference to 'cel_grammar::CelParser::SelectOrCallContext::member()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x1988): error: undefined reference to 'cel_grammar::CelParser::NegateContext::member()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x2fb0): error: undefined reference to 'cel_grammar::CelParser::CalcContext::unary()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x3bf0): error: undefined reference to 'cel_grammar::CelParser::CalcContext::calc(unsigned long)'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x4eb8): error: undefined reference to 'cel_grammar::CelParser::IndexContext::member()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x4f58): error: undefined reference to 'cel_grammar::CelParser::RelationContext::calc()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x58b0): error: undefined reference to 'cel_grammar::CelParser::LogicalNotContext::member()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x7980): error: undefined reference to 'cel_grammar::CelParser::StartContext::expr()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x85d8): error: undefined reference to 'cel_grammar::CelParser::MemberExprContext::member()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0x8940): error: undefined reference to 'cel_grammar::CelParser::PrimaryExprContext::primary()'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0xb0e8): error: undefined reference to 'cel_grammar::CelParser::RelationContext::relation(unsigned long)'
bazel-out/k8-opt/bin/external/com_google_cel_cpp/parser/_objs/visitor/visitor.o(.debug_addr+0xd3b8): error: undefined reference to 'cel_grammar::CelParser::ConstantLiteralContext::literal()'
collect2: error: ld returned 1 exit status
[4,995 / 4,995] checking cached actions
Target //source/exe:envoy-static failed to build
```

--- a/pkgs/servers/http/pomerium/default.nix
+++ b/pkgs/servers/http/pomerium/default.nix
@@ -7,7 +7,7 @@
 }:
 
 let
-  inherit (lib) concatStringsSep mapAttrsToList;
+  inherit (lib) concatStringsSep concatMap id mapAttrsToList;
 in
 buildGoModule rec {
   pname = "pomerium";
@@ -28,24 +28,38 @@ buildGoModule rec {
   ldflags = let
     # Set a variety of useful meta variables for stamping the build with.
     setVars = {
-      Version = "v${version}";
-      BuildMeta = "nixpkgs";
-      ProjectName = "pomerium";
-      ProjectURL = "github.com/pomerium/pomerium";
+      "github.com/pomerium/pomerium/internal/version" = {
+        Version = "v${version}";
+        BuildMeta = "nixpkgs";
+        ProjectName = "pomerium";
+        ProjectURL = "github.com/pomerium/pomerium";
+      };
+      "github.com/pomerium/pomerium/internal/envoy" = {
+        OverrideEnvoyPath = "${envoy}/bin/envoy";
+      };
     };
-    varFlags = concatStringsSep " " (mapAttrsToList (name: value: "-X github.com/pomerium/pomerium/internal/version.${name}=${value}") setVars);
+    concatStringsSpace = list: concatStringsSep " " list;
+    mapAttrsToFlatList = fn: list: concatMap id (mapAttrsToList fn list);
+    varFlags = concatStringsSpace (
+      mapAttrsToFlatList (package: packageVars:
+        mapAttrsToList (variable: value:
+          "-X ${package}.${variable}=${value}"
+        ) packageVars
+      ) setVars);
   in [
     "${varFlags}"
   ];
 
   preBuild = ''
+    # Replace embedded envoy with nothing.
+    # We set OverrideEnvoyPath above, so rawBinary should never get looked at
+    # but we still need to set a checksum/version.
     rm internal/envoy/files/files_{darwin,linux}*.go
     cat <<EOF >internal/envoy/files/files_generic.go
     package files
 
     import _ "embed" // embed
 
-    //go:embed envoy
     var rawBinary []byte
 
     //go:embed envoy.sha256
@@ -54,13 +68,9 @@ buildGoModule rec {
     //go:embed envoy.version
     var rawVersion string
     EOF
-    cp ${envoy}/bin/envoy internal/envoy/files/envoy
-    sha256sum ${envoy}/bin/envoy > internal/envoy/files/envoy.sha256
-    echo ${envoy.version} > internal/envoy/files/envoy.version
+    sha256sum '${envoy}/bin/envoy' > internal/envoy/files/envoy.sha256
+    echo '${envoy.version}' > internal/envoy/files/envoy.version
   '';
-
-  # We also need to set dontStrip to avoid having the envoy ZIP stripped off the end.
-  dontStrip = true;
 
   installPhase = ''
     install -Dm0755 $GOPATH/bin/pomerium $out/bin/pomerium

